akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -269,7 +269,7 @@ final class GroupBy[T, K](maxSubstreams: Int, keyFor: T ⇒ K) extends GraphStag
               subSubstreamSource.push(nextElementValue)
               clearNextElement()
             }
-          } else tryPull(in)
+          } else if (!hasBeenPulled(in)) tryPull(in)
```

Comment:
I think we could do these things still in 2.4.x, and keep the old ones deprecated. Maybe ticket worthy?